scripts/go_gen_p4_const.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022-present Open Networking Foundation
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ettle/strcase"
+	"github.com/golang/protobuf/proto"
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+)
+
+const (
+	P4infoPath = "conf/p4/bin/p4info.txt"
+
+	DefaultPackageName = "p4constants"
+
+	CopyrightHeader = "// SPDX-License-Identifier: Apache-2.0\n// Copyright 2022-present Open Networking Foundation\n\n"
+
+	ConstOpen  = "//noinspection GoSnakeCaseUsage\nconst (\n"
+	ConstClose = `)`
+
+	Uint32String = "uint32 = "
+
+	HfVarPrefix         = "Hdr_"
+	TblVarPrefix        = "Table_"
+	CtrVarPrefix        = "Counter_"
+	DirctrVarPrefix     = "DirectCounter_"
+	ActVarPrefix        = "Action_"
+	ActparamVarPrefix   = "ActionParam_"
+	ActprofVarPrefix    = "ActionProfile_"
+	PacketmetaVarPrefix = "PacketMeta_"
+	MtrVarPrefix        = "Meter_"
+)
+
+func generate(p4info *p4ConfigV1.P4Info, packageName *string) string {
```

Comment:
```suggestion
func generate(p4info *p4ConfigV1.P4Info, packageName string) string {
```

Doesn't have to be a pointer, we're not modifying it.